src/gui/sdlmain.cpp

```diff
@@ -1733,33 +1743,51 @@ void Config_Add_SDL() {
 	Pbool = sdl_sec->Add_bool("fullscreen",Property::Changeable::Always,false);
 	Pbool->Set_help("Start dosbox directly in fullscreen. (Press ALT-Enter to go back)");
 
-	Pbool = sdl_sec->Add_bool("fulldouble",Property::Changeable::Always,false);
-	Pbool->Set_help("Use double buffering in fullscreen. It can reduce screen flickering, but it can also result in a slow DOSBox.");
+	Pbool = sdl_sec->Add_bool("vsync", Property::Changeable::Always, false);
+	Pbool->Set_help("Sync to Vblank IF supported by the output device and renderer.\n"
+	                "It can reduce screen flickering, but it can also result in a slow DOSBox.");
 
-	Pstring = sdl_sec->Add_string("fullresolution",Property::Changeable::Always,"original");
+	Pstring = sdl_sec->Add_string("fullresolution", Property::Changeable::Always, "0x0");
 	Pstring->Set_help("What resolution to use for fullscreen: original, desktop or a fixed size (e.g. 1024x768).\n"
 	                  "Using your monitor's native resolution with aspect=true might give the best results.\n"
-			  "If you end up with small window on a large screen, try an output different from surface."
+	                  "If you end up with small window on a large screen, try an output different from surface."
 	                  "On Windows 10 with display scaling (Scale and layout) set to a value above 100%, it is recommended\n"
 	                  "to use a lower full/windowresolution, in order to avoid window size problems.");
 
 	Pstring = sdl_sec->Add_string("windowresolution",Property::Changeable::Always,"original");
 	Pstring->Set_help("Scale the window to this size IF the output device supports hardware scaling.\n"
 	                  "(output=surface does not!)");
 
-	const char* outputs[] = {
-		"surface", "overlay",
+	const char *outputs[] = {
+		"surface",
+		"texture",
+		"texturenb",
 #if C_OPENGL
-		"opengl", "openglnb",
+		"opengl",
+		"openglnb",
 #endif
-#if C_DDRAW
-		"ddraw",
-#endif
-		0 };
-	Pstring = sdl_sec->Add_string("output",Property::Changeable::Always,"surface");
+		0
+	};
+	Pstring = sdl_sec->Add_string("output",Property::Changeable::Always,"texture");
 	Pstring->Set_help("What video system to use for output.");
 	Pstring->Set_values(outputs);
 
+	const char *renderers[] = {
+		"auto",
+#ifdef WIN32
+		"direct3d",
+#endif
+		"opengl",
+#ifdef MACOSX
+		"metal",
+#endif
+		"software",
+		0
```

Comment:
More testing done. On ARM with OpenGL ES 2.0 support in the framebuffer (without X11), SDL2 will accept `opengles2` as the renderer (And DOSBox works!), if you can add it.  

Doing a bit more digging outside of testing, the Raspberry Pi also supports OpenGL ES 1.1, so we might as well add `opengles` too, as SDL2 also supports it.

https://wiki.libsdl.org/SDL_HINT_RENDER_DRIVER

Reply:
Test table updated with RPi results.
Performance wise, things were looking hopeless, but then I stumbled across a couple permutations that proved useable and fast.

Reply:
I totally blanked out, that OpenGL ES is completely missing from this table… I made two adjustments:

1) Renamed `renderer` to `texture_renderer` - to make it more understandable to the users. (I still don't like that `texture` and `texturenb` are separate outputs and not a separate pref, but this is out of scope here).
2) List of allowed values is now obtained via query to SDL - there's no point in hardcoding this information.

I think it's better than the current solution already, but not perfect (help text written in the conf file is not cross-platform and might confuse users). When unrecognized value is used (e.g. `direct3d` on Linux), then implementation seems to fall back to `auto` (but I don't know if this is dosbox doing it or SDL).

Reply:
> List of allowed values is now obtained via query to SDL - there's no point in hardcoding this information.
Huge improvement! If/when SDL adds more rendering back-ends, they will just show up for free now. 

I wonder if SDL2 offers a similar "query possible settings" function for its other video / audio / input-device configurable variables and bit-flags? 

I would be great to expose more of SDL2's "configurable" features to the users, and simply act as a pass-through for these (then check for an error and dump SDL2's error string to the user). This would let us eliminate some checks on our side, and let SDL2 become the gatekeeper on arguments, if they're conflicting or compatible, and so on.